test/controllers/api/messages_controller_test.rb

```diff
@@ -0,0 +1,611 @@
+        assert_select "[from_display_name='#{sender.display_name}']"
+        assert_select "[to_user_id='#{recipient.id}']"
+        assert_select "[to_display_name='#{recipient.display_name}']"
+        assert_select "[sent_on]"
+        assert_select "[deleted='#{!msg.to_user_visible}']"
+        assert_select "[message_read='#{msg.message_read}']"
+        assert_select "[body_format='markdown']"
+        assert_select "title", msg.title
+        assert_select "body", msg.body
+      end
+
+      # verify json output
+      get api_message_path(:id => msg.id, :format => "json"), :headers => recipient_auth
+      assert_equal "application/json", response.media_type
+      js = ActiveSupport::JSON.decode(@response.body)
+      jsm = js["message"]
+      assert_not_nil jsm
+      assert_equal msg.id, jsm["id"]
+      assert_equal sender.id, jsm["from_user_id"]
+      assert_equal sender.display_name, jsm["from_display_name"]
+      assert_equal recipient.id, jsm["to_user_id"]
+      assert_equal recipient.display_name, jsm["to_display_name"]
+      assert_equal msg.title, jsm["title"]
+      assert_not_nil jsm["sent_on"]
+      assert_equal msg.message_read, jsm["message_read"]
+      assert_equal !msg.to_user_visible, jsm["deleted"]
+      assert_equal "markdown", jsm["body_format"]
+      assert_equal msg.body, jsm["body"]
+
+      get api_message_path(:id => msg.id), :headers => sender_auth
+      assert_equal "application/xml", response.media_type
+      assert_select "message", :count => 1 do
+        assert_select "[id='#{msg.id}']"
+        assert_select "[from_user_id='#{sender.id}']"
+        assert_select "[from_display_name='#{sender.display_name}']"
+        assert_select "[to_user_id='#{recipient.id}']"
+        assert_select "[to_display_name='#{recipient.display_name}']"
+        assert_select "[sent_on]"
+        assert_select "[deleted='#{!msg.from_user_visible}']"
+        assert_select "[message_read='#{msg.message_read}']", 0
+        assert_select "[body_format='markdown']"
+        assert_select "title", msg.title
+        assert_select "body", msg.body
+      end
+
+      # verify json output
+      get api_message_path(:id => msg.id, :format => "json"), :headers => sender_auth
+      assert_equal "application/json", response.media_type
+      js = ActiveSupport::JSON.decode(@response.body)
+      jsm = js["message"]
+      assert_not_nil jsm
+      assert_equal msg.id, jsm["id"]
+      assert_equal sender.id, jsm["from_user_id"]
+      assert_equal sender.display_name, jsm["from_display_name"]
+      assert_equal recipient.id, jsm["to_user_id"]
+      assert_equal recipient.display_name, jsm["to_display_name"]
+      assert_equal msg.title, jsm["title"]
+      assert_not_nil jsm["sent_on"]
+      assert_equal !msg.from_user_visible, jsm["deleted"]
+      assert_not jsm.key?("message_read")
+      assert_equal "markdown", jsm["body_format"]
+      assert_equal msg.body, jsm["body"]
+    end
+
+    def test_read_status
```

Comment:
It would probably be better to call this `test_update_status` as the current name can be read two ways and I initially read it as testing that the status can be read rather than that the read status can be changed.